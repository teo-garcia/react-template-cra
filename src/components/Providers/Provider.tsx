import { BrowserRouter } from 'react-router-dom';
import { ProvidersProps } from '../../lib/types/client';

function Providers(props: ProvidersProps) {
  const { children } = props;
  return <BrowserRouter>{children}</BrowserRouter>;
}

export default Providers;
