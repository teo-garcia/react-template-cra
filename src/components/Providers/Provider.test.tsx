import { render, screen } from '@testing-library/react';
import Providers from './Provider';

describe('<Providers /> tests', function () {
  test('Should match with the component snapshot', function () {
    const { asFragment } = render(<Providers>Hello World</Providers>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Should render the provided children prop', function () {
    render(<Providers>Hello World</Providers>);
    expect(screen.getByText(/Hello World/i)).toBeInTheDocument();
  });
});
