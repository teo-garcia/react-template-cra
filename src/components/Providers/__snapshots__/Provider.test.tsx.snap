// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Providers /> tests Should match with the component snapshot 1`] = `
<DocumentFragment>
  Hello World
</DocumentFragment>
`;
